020-Getting-started/070-nextjs.mdx

```diff
@@ -50,185 +46,619 @@ apiKey=YOUR_API_KEY_HERE
+
+i Use xata pull main to regenerate code and types from your Xata database
+```
+
+</details>
 
-Your `.env` file should look something like this.
+On completion, the CLI will create `.env`, `.xatarc`, and `src/xata.ts` files within your project folder with the
+correct credentials to access your database.
 
-```bash
-# You should now have a ~/my-xata-project/.env file
+Your `.env` file should look something like this:
+
+```bash title=".env"
 XATA_API_KEY=YOUR_API_KEY_HERE
 XATA_BRANCH=main
 ```
 
-Since we asked for TypeScript support it also created files that provide typings and functions to call using Xata's TypeScript SDK. This will additionally be referenced in the `.xatarc` file as follows.
+Since we selected TypeScript support, it also created files that provide typings and functions to call using Xata's
+TypeScript SDK. This will additionally be referenced in the `.xatarc` file as follows:
 
 ```json
 {
-  "databaseUrl": "https://my-xata-app-database-url"
+  "databaseUrl": "https://xata-nextjs-database-url"
   "codegen": {
     "output": "src/xata.ts"
   }
 }
 ```
 
-The `src/xata.ts` includes generated code that typically you should never touch manually. If you change the schema of your `my-xata-app` database, like adding a column to `Posts`, you'll want to run `xata pull` to update the typings within that file.
+The `src/xata.ts` file includes generated code you should typically never touch manually.
+
+## Define the database schema and seed via CSV import
+
+You can use the [Xata UI][0] to manually define your schema and add data. However, for this guide, we'll walk through
+using the Xata CLI and a CSV file to:
+
+1.  Auto-generate a schema based on column headings for names and data types inferred from the column values
+2.  Seed the database with data
+
+**TODO: Update URL to CSV file**
+
+First, download the
+[example posts CSV file](https://raw.githubusercontent.com/leggetter/xata-examples/nextjs-getting-started/apps/getting-started-nextjs/seed/blog-posts.csv).
+You can either do this manually or by running the following command:
+
+```sh
+curl https://raw.githubusercontent.com/leggetter/xata-examples/nextjs-getting-started/apps/getting-started-nextjs/seed/blog-posts.csv --create-dirs -o seed/blog-posts.csv
+```
+
+Next, import the CSV:
+
+```sh
+xata import csv seed/blog-posts.csv --table Posts --create
+```
+
+<details>
+  <summary><code>xata import csv</code> output</summary>
+
+```sh
+xata import csv seed/blog-posts.csv --table Posts --create
+i 45 rows successfully imported 0 errors. 100% complete
+✔ Completed
+```
+
+</details>
+
+Now, if you open up the Xata UI and navigate to your database (you can also run `xata browse` to open up a browser
+window), you will see the **Posts** table:
+
+![Posts table](images/xata-table-posts.png)
+
+Click on the **Schema** link to see the schema definition with the interred data types:
```

Comment:
I think it was a typo from "inferred"?

Reply:
It might make sense to discuss briefly the `xata.*` metadata columns. They are automatically maintained by Xata and they are convenient when needed + they help for replicating to Elasticsearch.